Quorum/apis/git_api/git_manager.py

```diff
@@ -52,8 +62,8 @@ def clone_or_update(self) -> None:
         If the repository already exists locally, it will update the repository and its submodules.
         Otherwise, it will clone the repository and initialize submodules.
         """
-        for repo_name, repo_url in self.repos.items():
-            repo_path: Path = self.customer_path / repo_name
+        def clone_or_update_for_repo(repo_name: str, repo_url: str, to_path: Path):
```

Comment:
Maybe extract it outside instead of an inner function?
lambda functions are mostly 1 liners and their common usage is to be passed to high level functions like map etc.

Reply:
yeah no problem